text/006-bft-based-ordering-service.md

```diff
@@ -0,0 +1,404 @@
+
+The channel configuration contains two aspects: 
+
+ 1. Configuration for consenters, which would differ from what is [in SmartBFT](https://github.com/SmartBFT-Go/fabric/blob/release-2.3-bft/sampleconfig/configtx.yaml#L367-L397). 
+   Instead of configuring the consenters in the [opaque consensus metadata](https://github.com/hyperledger/fabric-protos/blob/main/orderer/configuration.proto#L22), 
+   the consenters would be defined in the [global channel configuration](https://github.com/hyperledger/fabric-protos/blob/main/common/configuration.proto#L35-L49).
+
+````
+message Consenter {
+    uint32 id = 1;
+    string host = 2;
+    uint32 port = 3;
+    string msp_id = 4;
+    bytes identity = 5;
+    bytes client_tls_cert = 6;
+    bytes server_tls_cert = 7;
+}
+
+// Orderers is encoded into the configuration transaction as a configuration item of type Chain
+// with a Key of "Orderers" and a Value of Orderers as marshaled protobuf bytes
+message Orderers {
+    repeated Consenter consenter_mapping = 1;
+}
+````
+ 2. Consensus specific [configuration](https://github.com/SmartBFT-Go/fabric-protos/blob/release-2.3-BFT/orderer/smartbft/configuration.proto#L32-L69) stored in the [opaque consensus metadata](https://github.com/hyperledger/fabric-protos/blob/main/orderer/configuration.proto#L22), which contains
+    information on how to instantiate the BFT library.
+
+
+
+## Upgrading from Raft
+
+Upgrading from Raft will only be allowed (by transaction rule enforcement in the admission path) on channels that are:
+
+- Already operating with V3.0 capability activated
+
+- Already have their orderers defined in the [global channel configuration](https://github.com/hyperledger/fabric-protos/blob/main/common/configuration.proto#L35-L49).
+
+The latter constraint is to ensure that the orderers can communicate amongst themselves and to minimize human error.
+
+From an operational perspective, the upgrade process from Raft to BFT will be as follows:
+
+1. The channel is put in [maintenance mode](https://hyperledger-fabric.readthedocs.io/en/release-2.4/kafka_raft_migration.html), to allow only administrative tasks and forbid regular transactions.
+
+2. A representative of the channel gathers signatures on a configuration update from sufficient organizations. The configuration update changes the [consensus type](https://github.com/hyperledger/fabric-protos/blob/main/orderer/configuration.proto#L20) to 'smartBFT', and sets the [metadata](https://github.com/hyperledger/fabric-protos/blob/main/orderer/configuration.proto#L22) to be empty.
+
+3. The configuration update is wrapped into a transaction and is sent to the ordering service.
+
+4. The Raft chain realizes the consensus type has changed, and the Raft chain goes into standby mode which only allows migration backward to BFT but nothing else.
```

Comment:
Don't you mean 'forward to BFT' rather than 'backward to BFT'?

Reply:
yes